packages/react-utilities/src/compose/types.ts

```diff
@@ -95,6 +105,10 @@ export type UnionToIntersection<U> = (U extends unknown ? (x: U) => U : never) e
  */
 export type PropsWithoutRef<P> = 'ref' extends keyof P ? (P extends unknown ? Omit<P, 'ref'> : P) : P;
 
+/**
+ * Takes a slots record and returns a mapping of slot names to shorthand props for each slot along with putting the
+ * shorthand props of the primary slot at the top level.
+ */
 export type ComponentProps<Shorthands extends ObjectShorthandPropsRecord, Primary extends keyof Shorthands = 'root'> =
```

Comment:
I had comments about this in the other PR https://github.com/microsoft/fluentui/pull/20891#discussion_r760703873 -- maybe revert it in this PR so it's tracked in one place? Same for the `ComponentState` comment.